code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+		/obj/item/weapon/stock_parts/manipulator/nano,
+		/obj/item/weapon/stock_parts/subspace/analyzer,
+		/obj/item/weapon/stock_parts/subspace/filter,
+		/obj/item/weapon/stock_parts/subspace/amplifier
+	)
+
+/obj/machinery/mind_machine/mind_machine_hub/RefreshParts()
+	var/manipCount = 0
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		if(istype(SP, /obj/item/weapon/stock_parts/manipulator))
+			manipCount += SP.rating
+	manipRating = manipCount
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/findConnections() //Finds and links the hub with the pods on either side
+	connectOne = findConnectOne()
+	if(connectOne)
+		connectOne.connectedHub = src
+		connectOne.podNumber = 1
+	connectTwo = findConnectTwo()
+	if(connectTwo)
+		connectTwo.connectedHub = src
+		connectTwo.podNumber = 2
+	if ((connectOne) && (connectTwo))
+		podsConnected = TRUE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/findConnectOne()
+	for(var/obj/machinery/mind_machine/mind_machine_pod/CO in orange(1,src))
+		if(CO != connectTwo)
+			return CO
+/obj/machinery/mind_machine/mind_machine_hub/proc/findConnectTwo()
+	for(var/obj/machinery/mind_machine/mind_machine_pod/CT in orange(1,src))
+		if(CT != connectOne)
+			return CT
+
+/obj/machinery/mind_machine/mind_machine_hub/Destroy()
+	podsConnected = null
+	currentlySwapping = null
+	lockedPods = FALSE
+	occupantScan = FALSE
+	occupantOne = null
+	occupantTwo = null
+	swapProgress = 0
+	if(connectOne || connectTwo || podsConnected)
+		connectOne.connectedHub = null
+		connectTwo.connectedHub = null
+		connectOne.podNumber = 0
+		connectTwo.podNumber = 0
+		connectOne = null
+		connectTwo = null
+	. = ..()
+
+/obj/machinery/mind_machine/mind_machine_hub/attackby(var/obj/item/A as obj, var/mob/user as mob)
+	..()
+	if(istype(A, /obj/item/bluespace_crystal/flawless))
+		user.drop_item(A, src)
+		bluespaceConduit += 300
+		to_chat(user, "The bluespace conduit flashes violently before calming to a glow.")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/bluespace_crystal/artificial))
+		user.drop_item(A, src)
+		bluespaceConduit += 1
+		to_chat(user, "The crystal is assimilated into the conduit!")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/bluespace_crystal))
+		user.drop_item(A, src)
+		bluespaceConduit += 3
+		to_chat(user, "The crystal is assimilated into the conduit!")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/device/soulstone))
+		if(soulShardSafety != FALSE)
+			to_chat(user, "That slot is full!")
+			return
+		if(A.contents.len)
+			to_chat(user, "The stone must be empty to function properly.") //prevents deleting players in stones
+			return
+		else
+			user.drop_item(A, src)
+			to_chat(user, "The stone is pulled into the machine. You can hear crunching and a quiet sizzle from inside.")
+			qdel(A)
+			soulShardSafety = TRUE
+
+/obj/machinery/mind_machine/mind_machine_hub/attack_hand(user as mob)
+	if(podsConnected == FALSE || connectOne == FALSE || connectTwo == FALSE)
+		podsConnected = FALSE
```

Comment:
might want to give the player some feedback before nuking their stray connections

Reply:
Honestly I would go with a button in the UI that is only shown if the connections aren't fulfilled.